world/stat_checks/check_maker.py

```diff
@@ -388,6 +388,123 @@ def _get_fail_rolls(self):
+        super().__init__(
+            character=character,
+            receivers=receivers,
+            stat=stat,
+            skill=skill,
+            rating=rating,
+            **kwargs,
+        )
+
+        self.retainer = retainer
+
+    def execute(self):
+        self.raw_roll = randint(1, 100)
+
+        # Get retainer's roll values; they don't have knacks
+        # so those are ignored here.
+        stat_val = self.get_roll_value_for_stat()
+        skill_val = self.get_roll_value_for_skill()
+        rating_val = self.get_roll_value_for_rating()
+
+        self.result_value = self.raw_roll + stat_val + skill_val - rating_val
+
+        self.natural_roll_type = self.check_for_crit_or_botch()
+        self.roll_result_object = RollResult.get_instance_for_roll(
+            self.result_value, natural_roll_type=self.natural_roll_type
+        )
+
+        self.result_message = self.roll_result_object.render(**self.get_context())
+
+    def get_roll_value_for_stat(self) -> int:
+        if not self.stat:
+            return 0
+
+        stat_val = self.retainer.dbobj.traits.get_stat_value(self.stat)
+        return StatWeight.get_weighted_value_for_stat(stat_val, not self.skill)
+
+    def get_roll_value_for_skill(self) -> int:
+        if not self.skill:
+            return 0
+
+        skill_val = self.retainer.dbobj.traits.get_skill_value(self.skill)
+        return StatWeight.get_weighted_value_for_skill(skill_val)
+
+    @property
+    def check_string(self) -> str:
+        if self.skill:
+            return f"{self.stat} and {self.skill} at {self.rating}"
+        return f"{self.stat} at {self.rating}"
+
+    @property
+    def roll_prefix(self) -> str:
+        return f"{self.character}'s retainer ({self.retainer.pretty_name}|n) checks {self.check_string}"
+
+    def announce_to_room(self):
+        notifier = RoomNotifier(
+            self.character,
+            self.character.location,
+            to_player=True,
+            to_gm=True,
+            to_staff=True,
+        )
+
+        notifier.generate()
+        notifier.notify(self.roll_message, options={"roll": True})
+
+    def get_context(self) -> dict:
+        crit = None
+        botch = None
+        if self.natural_roll_type:
+            if self.natural_roll_type.is_crit:
+                crit = self.natural_roll_type
+            else:
+                botch = self.natural_roll_type
+
+        try:
+            if self.retainer.name.count(",") >= 1:
+                short_name = self.retainer.name.split(",", 1)
+                short_name = short_name[0].strip()
+            elif self.retainer.name.count("-") >= 1:
+                short_name = self.retainer.name.split("-", 1)
+                short_name = short_name[0].strip()
+            else:
+                short_name = self.retainer.name
+        except (ValueError, IndexError):
+            short_name = self.retainer.name
+
+        return {
+            "retainer": short_name,
```

Comment:
same here - replace "retainer" with "character", and there's no need to change the template. Ideally, from the template's perspective, all it should know is that it's getting an object to represent who made the roll. Unless we need to present information differently based on their nature, I'd keep it decoupled